pkg/database/database_test.go

```diff
@@ -510,14 +511,264 @@ func TestTxByID(t *testing.T) {
 	_, err := db.TxByID(nil)
 	require.Error(t, ErrIllegalArguments, err)
 
-	for ind, val := range kvs {
-		txhdr, err := db.Set(&schema.SetRequest{KVs: []*schema.KeyValue{{Key: val.Key, Value: val.Value}}})
-		require.NoError(t, err)
-		require.Equal(t, uint64(ind+2), txhdr.Id)
-	}
+	txhdr1, err := db.Set(&schema.SetRequest{
+		KVs: []*schema.KeyValue{
+			{Key: []byte("key0"), Value: []byte("value0")},
+		},
+	})
+	require.NoError(t, err)
 
-	_, err = db.TxByID(&schema.TxRequest{Tx: uint64(1)})
+	txhdr2, err := db.ExecAll(&schema.ExecAllRequest{
+		Operations: []*schema.Op{
+			{
+				Operation: &schema.Op_Ref{
+					Ref: &schema.ReferenceRequest{
+						Key:           []byte("ref1"),
+						ReferencedKey: []byte("key0"),
+					},
+				},
+			},
+			{
+				Operation: &schema.Op_Kv{
+					Kv: &schema.KeyValue{
+						Key:   []byte("key1"),
+						Value: []byte("value1"),
+					},
+				},
+			},
+			{
+				Operation: &schema.Op_ZAdd{
+					ZAdd: &schema.ZAddRequest{
+						Set:   []byte("set1"),
+						Score: 10,
+						Key:   []byte("key1"),
+					},
+				},
+			},
+		},
+	})
+	require.NoError(t, err)
+
+	_, _, err = db.SQLExec(&schema.SQLExecRequest{Sql: "CREATE TABLE mytable(id INTEGER AUTO_INCREMENT, PRIMARY KEY id)"}, nil)
 	require.NoError(t, err)
+
+	_, ctx1, err := db.SQLExec(&schema.SQLExecRequest{Sql: "INSERT INTO mytable() VALUES ()"}, nil)
+	require.NoError(t, err)
+	require.Len(t, ctx1, 1)
+
+	txhdr3 := ctx1[0].TxHeader()
+
+	t.Run("values should not be resolved but digests returned in entries field", func(t *testing.T) {
+		tx, err := db.TxByID(&schema.TxRequest{Tx: txhdr1.Id})
+		require.NoError(t, err)
+		require.NotNil(t, tx)
+		require.Len(t, tx.Entries, 1)
+		require.Empty(t, tx.KvEntries)
+		require.Empty(t, tx.ZEntries)
+
+		for _, e := range tx.Entries {
+			require.Len(t, e.Value, 0)
```

Comment:
We could also check here (and simillar checks below) if digest is not empty (or it's of the sha256 size):

```suggestion
			require.Len(t, e.Value, 0)
			require.Len(t, e.HValue, sha256.Size)
```

could even check if the sha256 is correct